src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java

```diff
@@ -26,4 +32,18 @@ public TransactionStatusResponse createCirculationRequest(String dcbTransactionI
     };
   }
 
+  @Override
+  public TransactionStatusResponse updateTransactionStatus(String dcbTransactionId, TransactionStatus transactionStatus) {
+    return transactionRepository.findById(dcbTransactionId).map(dcbTransaction -> {
+      if (Objects.requireNonNull(dcbTransaction.getRole()) == DcbTransaction.RoleEnum.LENDER) {
```

Comment:
I think lets also check if the current transaction status and the new transaction status is different then invoke the update transaction status.